### DISCUSSION TITLE
Smaller Zoom In/Out Steps for Ctrl+Scroll

### DISCUSSION THREAD

**Discussion by Foxxsayy** [freeplane/freeplane#1670] [Show and tell]
### The Issue this solves:
Most mice default to scrolling 3 lines at a time, because 1 line at a time is too slow for most things. Freeplane interprets a mouse scroll as "Zoom in X times," where X is the number of lines your mouse is set to scroll to. This results in a huge jump, which is annoying when navigating Freeplane with the mouse and the CTRL key. 

### What it does:
**Thus, I made an AutoHotKey script that makes 1 scroll on your mouse only 1 line at a time.**  So now you have small zoom steps on your mouse wheel!  Get it here: [Scroll Only 1 Line When Freeplane is Top Window.zip](https://github.com/freeplane/freeplane/files/14116881/Scroll.Only.1.Line.When.Freeplane.is.Top.Window.zip)

If you don't have AHK, I used [AHK2exe](https://github.com/AutoHotkey/Ahk2Exe) to compile it to an exe as well. Since AHK scripts are human readable, you can also check the script and compile it yourself, AHK2exe is super simple. 
[Scroll Only 1 Line When Freeplane is Top Window.zip]

### Details for nerds:
It works by changing the lines your mouse scrolls to to "1" anytime that Freeplane is in the foreground (specifically, anytime _javaw.exe_ is in the foreground, which is the .exe name for the Freeplane window that actually displays the mind map & editor).

[Scroll Only 1 Line When Freeplane is Top Window.zip](https://github.com/freeplane/freeplane/files/14116881/Scroll.Only.1.Line.When.Freeplane.is.Top.Window.zip)

**Comment by euu2021** [comment]
Interesting!

Now, one idea is to limit the scope of the script only to the mindmap frame. Doing this, for example:
`#If WinActive("ahk_exe javaw.exe") && WinActive("ahk_class SunAwtFrame")`

This way, the user keeps his usual scroll settings in the dialogs (for example, in the `Edit->Node core->Edit node core in dialog`).

**Comment by dpolivaev** [comment]
I'll fix it in Freeplane: it should always increase / decrease the zoom factor by 10% independently from the numeric value it gets from the mouse wheel
